client/lib/analytics/ad-tracking/setup.js

```diff
@@ -19,64 +8,66 @@ import {
 } from './constants';
 
 if ( typeof window !== 'undefined' ) {
-	setupGtag();
+	if ( mayWeTrackByTracker( 'ga' ) ) {
+		setupGtag();
+	}
 
 	// Facebook
-	if ( isFacebookEnabled ) {
+	if ( mayWeTrackByTracker( 'facebook' ) ) {
 		setupFacebookGlobal();
 	}
 
 	// Bing
-	if ( isBingEnabled && ! window.uetq ) {
+	if ( mayWeTrackByTracker( 'bing' ) && ! window.uetq ) {
 		window.uetq = [];
 	}
 
 	// Criteo
-	if ( isCriteoEnabled && ! window.criteo_q ) {
+	if ( mayWeTrackByTracker( 'criteo' ) && ! window.criteo_q ) {
 		window.criteo_q = [];
 	}
 
 	// Quantcast
-	if ( isQuantcastEnabled && ! window._qevents ) {
+	if ( mayWeTrackByTracker( 'quantcast' ) && ! window._qevents ) {
 		window._qevents = [];
 	}
 
 	// Google Ads Gtag for wordpress.com
-	if ( isWpcomGoogleAdsGtagEnabled ) {
+	if ( mayWeTrackByTracker( 'googleAds' ) ) {
```

Comment:
When I have the cookie that disables google analytics: `document.cookie = 'sensitive_pixel_options=%7B%22version%22%3A20201224%2C%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Afalse%2C%22advertising%22%3Atrue%7D%7D';`
I am getting an error thrown here. It seems like `mayWeTrackByTracker( 'googleAds' )` is still returning true even though the cookie has analytics disabled

Reply:
Here is the error:
![image](https://user-images.githubusercontent.com/65001528/198150535-a3dace9f-2591-4f33-af31-5a71995e739c.png)

Reply:
FWIW, this does completely stop the page from being loaded for me. I have country_code set to 'US' and region set to 'california'

Reply:
Thanks, @CodeyGuyDylan for spotting that one out, it appears the floodlight and google ads setup were incomplete. It should be fixed with the following commit!